### PR TITLE
feat: add spaced project name

### PR DESCRIPTION
**Pull request by JaeAeich**
#### Description

- Add support for human readable project name/title that may include uppercase characters and whitespace

#### Checklist

- [x] My code follows the [contributing guidelines][contributing-guidelines] of this
    project, including, in particular, with regard to any style guidelines
- [x] The title of my PR complies with the [Conventional Commits
    specification][conv-commits]; in particular, it clearly indicates
    that a change is a breaking change
- [x] I acknowledge that all my commits will be squashed into a single commit,
    using the PR title as the commit message
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have updated the user-facing documentation to describe any new or
    changed behavior
- [x] I have added type annotations for all function/class/method interfaces
    or updated existing ones (only for Python, TypeScript, etc.)
- [x] I have provided appropriate documentation ([Google-style Python
    docstrings][py-doc-google]) for all packages/modules/functions/classes/
    methods or updated existing ones
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature
    works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage

#### Comments

[contributing-guidelines]: https://elixir-cloud-aai.github.io/guides/guide-contributor/workflow/
[conv-commits]: https://www.conventionalcommits.org/en
[py-doc-google]: https://google.github.io/styleguide/pyguide.html

## Summary by Sourcery

Revise project naming conventions across the codebase to support spaced, dashed, and underscored formats, ensuring consistency in documentation, CI workflows, and test paths.

Enhancements:
- Update project naming conventions to use spaced, dashed, and underscored formats for better readability and consistency.

CI:
- Modify CI workflow scripts to use the new underscored project name format.

Documentation:
- Update README badges and links to reflect the new project naming conventions.

Tests:
- Adjust test paths to accommodate the new underscored project name format.

### CODE REVIEW

File: {{ cookiecutter.project_name_dashed }}/{{ cookiecutter.project_name_underscored }}/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-"""This package contains module for the {{ cookiecutter.project_slug }}."""
+"""This package contains module for the {{ cookiecutter.project_name_underscored }}."""
```
Comment (uniqueg) [inline {{ cookiecutter.project_name_dashed }}/{{ cookiecutter.project_name_underscored }}/__init__.py:1]: Better:

> Main package for {{ cookiecutter.project_name_underscored }}

**Review by uniqueg — CHANGES_REQUESTED** [review]
Maybe make this change in the Makefile as well in this PR:

https://github.com/elixir-cloud-aai/cloud-storage-handler/pull/17#discussion_r1730066163

Rest looks good

**Comment by JaeAeich** [comment]
> Maybe make this change in the Makefile as well in this PR:
> 
> [elixir-cloud-aai/cloud-storage-handler#17 (comment)](https://github.com/elixir-cloud-aai/cloud-storage-handler/pull/17#discussion_r1730066163)

My reasoning was to let the user/dev know that, the corresponding console script `project_name_dashed` has been installed in their environment too. Well I have made that changes, if you think otherwise let me know.

**Comment by JaeAeich** [comment]
Please check that the [package](https://test.pypi.org/project/after-rename-done/) meets all the requirements and lets merge.

image

File: hooks/pre_gen_project.py [hunk 2]
```diff
@@ -2,9 +2,9 @@
 import sys
 
 MODULE_REGEX = r'^[_a-zA-Z][_a-zA-Z0-9]+$'
-MODULE_NAME = '{{ cookiecutter.project_slug }}'
+MODULE_NAME = '{{ cookiecutter.project_name_underscored }}'
 PROJECT_REGEX = r'^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*$'
-PROJECT_NAME = '{{ cookiecutter.project_name }}'
+PROJECT_NAME = '{{ cookiecutter.project_name_dashed }}'
```
[the remark this answers is not in the document]
Comment (JaeAeich) [reply hooks/pre_gen_project.py:7]: Can you please suggest some, this seems correct to me :sweat_smile:. underscore is is then name of the module and dash is the name of the project.

**Comment by JaeAeich** [comment]
Also this is the file str after. 

`tree` command ran  inside the dir `after-rename-done`.

```
.
├── after_rename_done
│   ├── __init__.py
│   └── main.py
├── dist
│   ├── after_rename_done-0.1.0-py3-none-any.whl
│   └── after_rename_done-0.1.0.tar.gz
├── docs
│   ├── make.bat
│   ├── Makefile
│   └── source
│       ├── conf.py
│       ├── index.rst
│       └── pages
│           └── modules.rst
├── images
│   ├── logo-elixir-cloud-aai.svg
│   └── logo-elixir.svg
├── LICENSE
├── Makefile
├── poetry.lock
├── PULL_REQUEST_TEMPLATE.md
├── pyproject.toml
├── README.md
└── tests
    ├── __init__.py
    ├── test_integration
    │   ├── __init__.py
    │   └── test_init.py
    └── test_unit
        ├── __init__.py
        └── test_init.py

10 directories, 22 files
```

File: hooks/pre_gen_project.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (uniqueg) [reply hooks/pre_gen_project.py:7]: Well, according to the names of our input variables for Cookiecutter, `project_name` is the human-readable name of the project and what you call `PROJECT_NAME` here is `project_name_dashed`.

But I think it's no big deal, it's fine - it's not a generated file anyway.

**Review by uniqueg — APPROVED** [review]

**Pull request merged**